src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -181,21 +181,22 @@ private void loadMetricConfigFiles(AppConfig appConfig, LinkedList<Configuration
      * Format the instance tags defined in the YAML configuration file to a `LinkedHashMap`.
      * Supported inputs: `List`, `Map`.
      */
-    private static LinkedHashMap<String, String> getTagsMap(Object tagsMap){
-        try {
-            // Input has `Map` format
-            return (LinkedHashMap<String, String>) tagsMap;
+    private static LinkedHashMap<String, String> getTagsMap(Object tagsMap, AppConfig appConfig) {
+        LinkedHashMap<String, String> tags = new LinkedHashMap<String, String>();
+        if (appConfig.getGlobalTags() != null) {
+            tags.putAll(appConfig.getGlobalTags());
         }
-        catch (ClassCastException e){
-            // Input has `List` format
-            LinkedHashMap<String, String> tags = new LinkedHashMap<String, String>();
-
-            for (String tag: (List<String>)tagsMap) {
-                tags.put(tag, null);
+        if (tagsMap != null) {
+            try {
+                // Input has `Map` format
```

Comment:
Nit: I guess you could have left `// Input has `List` format`.

Reply:
Updated